internal/runtime/client/client.go

```diff
@@ -75,6 +76,12 @@ type Client interface {
 
 	// Unregister unregisters the ExtensionConfig.
 	Unregister(extensionConfig *runtimev1.ExtensionConfig) error
+
+	// CallAll calls all the extension registered for the hook.
+	CallAll(ctx context.Context, hook catalog.Hook, request runtime.Object, response runtimehooksv1.AggregatableResponse) error
+
+	// Call calls only the extension with the given name.
+	Call(ctx context.Context, name string, request, response runtime.Object) error
```

Comment:
How about 
```go
// CallHook calls all the extensions registered for the hook.
CallHook(ctx context.Context, hook catalog.Hook, request runtime.Object, response runtimehooksv1.AggregatableResponse) error

// CallExtension calls the extension with the given name.
CallExtension(ctx context.Context, name string, request, response runtime.Object) error
```

How about using `CallHook` and `CallExtension` or (`CallByHook` and `CallByExtension`) instead of `CallAll` and `Call`?

Once we remove the builder pattern the usage is going to be `runtimeClient.CallAll()` which seems unclear and ambiguous.

Reply:
I would suggest: `CallExtensions` (or `CallAllExtensions`) and `CallExtension`.

I would like to avoid mixing call hook and call extension, as a user I would wonder what the difference between a hook and an extension is.

I wonder if it would be better if we also add catalog.Hook as a parameter to `CallExtension`. I know we don't strictly need it, but it might be better if a user has to explicitly declare what hook they want to call and we use that information to make sure that the type matches what is in the registry (i.e. registration.GroupVersionHook).

@fabriziopandini We discussed it and for now we would like to avoid mixing top-level discovery calls and extension calls on a deeper layer. I think it's better to keep it all flat at this point and not mix patterns and then potentially change it when we reconsider this later.